torch_xla/csrc/ops/dynamic_ir.cpp

```diff
@@ -0,0 +1,75 @@
+#include "torch_xla/csrc/ops/dynamic_ir.h"
+
+#include "absl/strings/str_join.h"
+#include "tensorflow/compiler/xla/xla_client/debug_macros.h"
+#include "torch_xla/csrc/lowering_context.h"
+#include "torch_xla/csrc/ops/infer_output_shape.h"
+
+namespace torch_xla {
+
+DimensionNode::DimensionNode(torch::lazy::OpKind op, OpList operands,
+                             torch::lazy::hash_t hash_seed)
+    : XlaNode(op, operands, xla::ShapeUtil::MakeShape(xla::S32, {}),
+              /*num_outputs=*/1,
+              /* hash_seed */ hash_seed) {}
+
+std::string DimensionNode::ToString() const { return "DimensionNode"; }
+
+SizeNode::SizeNode(XlaValue input, size_t dim)
+    : DimensionNode(
+          torch::lazy::OpKind{c10::Symbol::fromQualString("aten::size")},
+          {input}, torch::lazy::MHash(dim)),
+      dim_(dim){};
+
+XlaOpVector SizeNode::Lower(LoweringContext* loctx) const {
+  auto input = loctx->GetOutputOp(operand(0));
+  return ReturnOp(xla::GetDimensionSize(input, this->dim_), loctx);
+}
+
+int64_t SizeNode::getStaticValue() const {
+  return dynamic_cast<const XlaNode*>(operand(0).node)->shape(0).size(dim_);
+}
+
+std::string SizeNode::ToString() const { return "SizeNode"; }
+
+SizeAdd::SizeAdd(XlaValue a, XlaValue b)
```

Comment:
should we check that XlaValue actually contains `DimensionNode` and store them as `DimensionNode` diretly? This can save us from doing multiple `dynamic_cast` on `getStaticValue`.

Reply:
I am flexible. @Krovatkin wdyt?